### PR TITLE
Raspberry Pi changes

### PR DESCRIPTION
**Pull request by marschap**
Hi,

some fixes & extensions for Raspberry Pi:
* support multiple displays with hd44780-gpio
* correctly detect Raspberry Pi's in newer kernels too
* always initialize hw variable in hd44780-rpi

Please consider integrating them in the master repo, as they allow better Raspberry support with LibreELEC.

Thanks in advance
Peter

### CODE REVIEW

**Comment by marschap** [comment]
BTW, commit "	hd44780-gpio: support multiple displays" should fix issue #54

File: server/drivers/hd44780-gpio.c [hunk 1]
```diff
@@ -159,12 +175,12 @@ hd_init_gpio(Driver *drvthis)
...
-	send_nibble(p, (FUNCSET | IF_8BIT) >> 4);
+	send_nibble(p, (FUNCSET | IF_8BIT) >> 4, 0);
 	p->hd44780_functions->uPause(p, 4100);
-	send_nibble(p, (FUNCSET | IF_8BIT) >> 4);
+	send_nibble(p, (FUNCSET | IF_8BIT) >> 4, 0);
 	p->hd44780_functions->uPause(p, 100);
-	send_nibble(p, (FUNCSET | IF_8BIT) >> 4);
-	send_nibble(p, (FUNCSET | IF_4BIT) >> 4);
+	send_nibble(p, (FUNCSET | IF_8BIT) >> 4, 0);
+	send_nibble(p, (FUNCSET | IF_4BIT) >> 4, 0);
```
Comment (haraldg) [inline server/drivers/hd44780-gpio.c:184]: I think you are missing initialization of the second display here?
(I don't know anything about the protocol or the initialization procedure - just applying common sense, so maybe I'm wrong.)

**Review by haraldg — COMMENTED** [review]
Thanks, for your changes - especially the one for the gpio subdriver. Aside from the one comment I think they are fine.

BTW: Did you have a look at #49 ?

**Comment by marschap** [comment]
Hi,

Am Dienstag, 28. März 2017, 09:32:42 CEST schrieb haraldg:
> Thanks, for your changes - especially the one for the gpio subdriver. Aside
> from the one comment I think they are fine.
> 
> BTW: Did you have a look at #49 ?
No, I simply took over what worked with the hd44780-rpi driver.

> >  	p->hd44780_functions->uPause(p, 100);
> 
> -	send_nibble(p, (FUNCSET | IF_8BIT) >> 4);
> -	send_nibble(p, (FUNCSET | IF_4BIT) >> 4);
> +	send_nibble(p, (FUNCSET | IF_8BIT) >> 4, 0);
> +	send_nibble(p, (FUNCSET | IF_4BIT) >> 4, 0);
> 
> I think you are missing initialization of the second display here?
> (I don't know anything about the protocol or the initialization procedure -
> just applying common sense, so maybe I'm wrong.)
No. the logic for talking to the displays in the drivers is
*   0    => all displays
*   X > 0 => display X
The "magic" for that is inside send_nibble().

Best
Peter

-- 
Peter Marschall
peter.marschall@adpm.de

**Pull request merged**